showcase/treemap/tree-tools.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Find branch in tree by omen pair - key-value
+ * @param {Object} omen - how to find branch. Example - { id: 123456 }
+ * @param {Object} options [{children: 'children', returnChild: true}];
+ * @param {Boolean} options.returnChild -
+ * if true return branch started from found child,
+ * else return whole branch where child founded
+ */
+export function findBranchByOmen(omen, options) {
+  const [[omenProp, omenValue]] = Object.entries(omen);
+  const opt = {
+    children: 'children',
+    returnChild: true,
+    ...options
+  };
+
+  function _findBranch(tree) {
+    if (omenValue === tree[omenProp]) return tree;
+    if (!tree[opt.children]) return false;
+    return opt.returnChild
+      ? tree[opt.children].reduce((result, branch) => _findBranch(branch) || result, false)
+      : tree[opt.children].find(branch => _findBranch(branch));
+  }
+
+  return data => _findBranch(data);
+}
+
+/**
+ * Fold childrens, and recalculate weights on leafs
+ * @param {Object} tree
+ * @param {Object} options [{
+ *    children: 'children',
+ *    value: 'value',
+ *    maxLevel: 2,
+ *    clearValues: false - if true clear values from branchs (if they not leaf)
+ *  }]
+ */
+export function foldChilndrens(tree, options, level = 0) {
```

Comment:
typo

Reply:
Done. I added types and comments in 5cba103702621303f48a77de652403ad560052df